react/features/virtual-background/components/VirtualBackgroundPreview.js

```diff
@@ -0,0 +1,270 @@
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            trackData: {},
+            loading: false,
+            selectedTrack: null
+        };
+    }
+
+    /**
+     * Creates and updates the track data.
+     *
+     * @returns {void}
+     */
+    async _setTracks() {
+        this._disposeTracks(this.state.trackData);
+
+        const [ trackData ] = await createLocalTracksF({
+            cameraDeviceId: this.props._currentCameraDeviceId,
+            devices: [ 'video' ]
+        });
+
+
+        // In case the component gets unmounted before the tracks are created
+        // avoid a leak by not setting the state
+        if (this._componentWasUnmounted) {
+            this._disposeTracks(trackData);
+        } else {
+            this.setState({
+                trackData
+            });
+        }
+
+        // background preview should open the selected device camera only
+        const selectedTrack = this.state.trackData;
+
+        this.setState({
+            selectedTrack
+        });
+        if (this.props._virtualBackground?.backgroundEffectEnabled && this.state.selectedTrack) {
+            await this.state.selectedTrack.setEffect(
+                await createVirtualBackgroundEffect(this.props._virtualBackground)
+            );
+        }
+    }
+
+    /**
+     * Destroys all the tracks from trackData object.
+     *
+     * @param {Object} trackData - An array of tracks that are to be disposed.
+     * @returns {Promise<void>}
+     */
+    _disposeTracks(trackData) {
+        // eslint-disable-next-line no-unused-expressions
+        Object.keys(trackData).length !== 0 && trackData.disposed === true;
+    }
+
+    /**
+     * Apply background effect on video preview.
+     *
+     * @returns {Promise}
+     */
+    async _applyBackgroundEffect() {
+        this.setState({ loading: true });
+        await this.props.dispatch(toggleBackgroundEffect(this.props.options, this.state.selectedTrack));
+        this.setState({ loading: false });
+    }
+
+    /**
+     * Apply video preview loader.
+     *
+     * @returns {Promise}
+     */
+    _loadVideoPreview() {
+        return (
+            <div className = 'video-preview-loader'>
+                <Spinner
+                    invertColor = { true }
+                    isCompleting = { false }
+                    size = { 'large' } />
+            </div>
+        );
+    }
+
+    /**
+     * Renders a preview entry.
+     *
+     * @param {Object} data - The track data.
+     * @returns {React$Node}
+     */
+    _renderPreviewEntry(data) {
+        const { error } = data;
+        const { t } = this.props;
+        const className = 'video-background-preview-entry';
+
+        if (this.state.loading) {
+            return this._loadVideoPreview();
+        }
+        if (error) {
```

Comment:
Can you show a screenshot of this?

Reply:
<img width="822" alt="Screenshot 2021-05-05 at 19 22 37" src="https://user-images.githubusercontent.com/18324739/117176291-65f49100-add8-11eb-9646-1fba3bb68ef3.png">